app/controllers/superpowers_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SuperpowersController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create]
+
+  def index
+    @superpowers = Superpower.all
+  end
+
+  def show
+    @superpower = Superpower.find(params[:id])
+  end
+
+  def new
+    @superpower = Superpower.new
+  end
+
+  def create
+    @superpower = Superpower.new(superpower_params)
+    @superpower.user = current_user
+
+    if @superpower.save
+      flash[:notice] = 'Superpower added successfully'
+      redirect_to superpower_path(@superpower)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @superpower = Superpower.find(params[:id])
+  end
+
+  def update
```

Comment:
in a separate / new feature branch, you want to assert that the current user has the authorization to update or delete a superpower